sdk/search/Azure.Search.Documents/src/Indexes/Models/VectorSearchField.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    /// <summary>
+    /// A searchable vector field of type "Collection(<see cref="SearchFieldDataType.Single"/>)".
+    /// </summary>
+    public class VectorSearchField : SearchFieldTemplate
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SimpleField"/> class.
+        /// </summary>
+        /// <param name="name">The name of the field, which must be unique within the index or parent field.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        public VectorSearchField(string name) : base(name, SearchFieldDataType.Collection(SearchFieldDataType.Single))
+        {
+        }
+
+        /// <summary> The dimensionality of the vector field. </summary>
+        public int? VectorSearchDimensions { get; set; }
```

Comment:
Isn't the dimensionality required for vectorized search fields

Reply:
It isn't required in the swagger for SearchField. Should we make `VectorSearchDimensions` and `VectorSearchProfileName` required for `VectorSearchField`? Did you make these required in java?

Reply:
It isn't required in the swagger for SearchField. Should we make VectorSearchDimensions and VectorSearchProfileName required for VectorSearchField? Did you make these required in java?

Reply:
Made both the properties required now that we have a separate type for vector fields.